linux/example/flutter_embedder_example.cc

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <cstdlib>
+#include <iostream>
+
+#include <flutter_embedder.h>
+
+int main(int argc, char **argv) {
+  // Edit this as needed.
+  std::string flutter_example_root = "../example_flutter";
+  std::string flutter_git_root = "../../flutter";
+  if (!glfwInit()) {
+    std::cout << "Couldn't init GLFW";
+  }
+  // Arguments for the Flutter Engine.
+  int arg_count = 1;
+  const char *args_arr[] = {
```

Comment:
The embedder expects arg[0] to be the binary as with normal command line parsing, so the --dart-non-checked-mode will be ignored here.

I'm not sure whether the engine actually cares what the first argument (passing "" might be fine), but for now on macOS we pass the actual arg[0] through in the example, and we should just go ahead and match.

Reply:
Fixed!